config/serialization.go

```diff
@@ -24,26 +27,30 @@ func configFile() string {
 			configFolder = filepath.Join(homeFolder, ".config")
 		}
 	}
-	return filepath.Join(configFolder, "taskcluster.yml")
+	configFileString := filepath.Join(configFolder, "taskcluster.yml")
+	return strings.NewReader(configFileString), configFileString
 }
 
 // Load will load confiration file, and initialize a default configuration
 // if no configuration is present. This only returns an error if a configuration
 // file is present, but we are unable to parse it.
 // TODO	we could simplify this function and only go through the definitions once
 //		and what happens if a value is tied to an env var AND to the config file?
-func Load() (map[string]map[string]interface{}, error) {
+func Load(configFileReader io.Reader) (map[string]map[string]interface{}, error) {
```

Comment:
So I would separate out configFile() into two functions, one that returns the string location and one that returns the io.Reader. Then in init.go and other calls for Load I can lose that extra step of extracting the io.Reader that I need. That's awesome. 

But then if I revert to Load without parameters, what would my testing look like, without being able to call from my test file and pass in the fake config data? I'd originally tried something like this using an interface, but discussing with @walac, this seemed the way to go instead.